src/Utilities/_DataValidation.php

```diff
@@ -59,3 +59,41 @@ function validateOperationTimeout(?int $operationTimeout=null)
         }
     }
 }
+
+if (!function_exists('validateTruncateSize'))
+{
+    function validateTruncateSize(?int $truncateSize=null)
+    {
+        if ($truncateSize === null)
+        {
+            return;
+        }
+        if ($truncateSize <= 0)
+        {
+            throw new InvalidArgumentError("Truncate size must be greater than zero.");
+        }
+    }
+}
+
+if (!function_exists('validateRange'))
+{
+    function validateRange(?int $beginIndex, ?int $count)
```

Comment:
Do we want to prevent users from passing `count` value larger than the actual size of the list??

Reply:
Validating that would require us to make a call to get the length of the list, and we don't want to make a round trip to the cache server for client side validation in my opinion. A number larger than the size of the list just deletes the rest of the list from the offset without complaining, so that round trip wouldn't buy us anything anyway. If you strongly disagree (or anyone else reading this disagrees) let me know and we can talk about it. 

That's a good thing to add to the tests in case the behavior ever changes . . . I'll do that.

Reply:
As I wrote this question, I was thinking the same thing about calling another API to get length will add latency so that's probably not something we want to do.
But I didn't know the behavior you explained when a number larger than the size of the list is passed, and I think it makes sense.